python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -452,6 +456,9 @@ def addDataset(self, datasetType, dataId, run, producer=None, recursive=False, *
                                                                        quantum_id=None,
                                                                        **links))
         datasetRef._id = result.inserted_primary_key[0]
+        # If the result is reported as a list of a number, unpack the list
+        if isinstance(datasetRef._id, list):
+            datasetRef._id = datasetRef._id[0]
```

Comment:
Do you need this for Oracle? This seems to contradict sqlalchemy documentation: https://docs.sqlalchemy.org/en/latest/core/connections.html?highlight=inserted_primary_key#sqlalchemy.engine.ResultProxy.inserted_primary_key
(we should probably report it if Oracle backend violates that)

Reply:
Yes, what is returned is a list on an element and an exception is raised

Reply:
I'd like to spend a little bit of time on understanding that when I'm back, OK to have this for now.